MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/BuyClient.java

```diff
@@ -245,7 +247,7 @@ public void setPageSize(int pageSize) {
      * @param callback the {@link Callback} that will be used to indicate the response from the asynchronous network operation, not null
      */
     public void getShop(final Callback<Shop> callback) {
-        retrofitService.getShop().subscribe(new InternalCallback<Shop>() {
+        retrofitService.getShop().observeOn(AndroidSchedulers.mainThread()).subscribe(new InternalCallback<Shop>() {
```

Comment:
Lets extract AndroidSchedulers.mainThread() as local field so in future we can configure what thread/scheduler client want to use as for call back.